src/Service/Import/GridAffectation/GridAffectationLoader.php

```diff
@@ -28,41 +35,169 @@ public function __construct(
+                .$item[GridAffectationHeader::USER_LASTNAME].', partenaire '.$item[GridAffectationHeader::PARTNER_NAME_INSTITUTION];
+            } else {
+                // email must be valid and not used by another user of another partner
+                $violations = $this->validator->validate($emailUser, $emailConstraint);
+                if (\count($violations) > 0) {
+                    $errors[] = 'Email incorrect pour un utilisateur : '.$emailUser;
+                }
+
+                /** @var User $userToCreate */
+                $userToCreate = $this->userManager->findOneBy(['email' => $emailUser]);
+
+                if (null !== $userToCreate && !\in_array('ROLE_USAGER', $userToCreate->getRoles())) {
+                    $errors[] = 'Il existe déjà un utilisateur avec cette adresse mail : '
+                    .$emailUser.' dans le territoire '.$userToCreate->getTerritory()->getName()
+                    .' et dans le partenaire '.$userToCreate->getPartner()->getNom()
+                    .' avec le rôle '.$userToCreate->getRoleLabel();
+                }
+                // store user mail to check duplicates
+                $mailUsers[] = $item[GridAffectationHeader::USER_EMAIL];
+            }
+            // user must have correct Role
+            if (!\in_array($item[GridAffectationHeader::USER_ROLE], array_keys(User::ROLES))) {
+                $errors[] = 'Rôle incorrect pour '.$item[GridAffectationHeader::USER_EMAIL].' --> '.$item[GridAffectationHeader::USER_ROLE];
+            }
+        }
+
+        // check if there are no duplicate email between partners
+        $occurrencesMailPartners = array_count_values($mailPartners);
+        $duplicatesMailPartners = array_filter($occurrencesMailPartners, function ($value) {
+            return $value > 1;
+        });
+        if (\count($duplicatesMailPartners) > 0) {
+            $errors[] = 'Certains partenaires ont un mail en commun '.implode(',', array_keys($duplicatesMailPartners));
+        }
+
+        // check if there are no duplicate email between users
+        $occurrencesMailUsers = array_count_values($mailUsers);
+        $duplicatesMailUsers = array_filter($occurrencesMailUsers, function ($value) {
+            return $value > 1;
+        });
+        if (\count($duplicatesMailUsers) > 0) {
+            $errors[] = 'Certains utilisateurs ont un mail en commun '.implode(',', array_keys($duplicatesMailUsers));
+        }
+
+        // check if there are no duplicate email between partners and users
+        $mails = array_merge($mailPartners, $mailUsers);
+        $occurrencesMails = array_count_values($mails);
+        $duplicatesMails = array_filter($occurrencesMails, function ($value) {
+            return $value > 1;
+        });
+        if (\count($duplicatesMails) > 0) {
+            $errors[] = 'Certains utilisateurs ont un mail en commun avec un partenaire '.implode(',', array_keys($duplicatesMails));
```

Comment:
méthode privée au lieu du commentaire `        // check if there are no duplicate email between partners and users
`